sale_quotation_capable_vendor_rfq/models/sale_order.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    purchase_order_ids = fields.One2many(
+        "purchase.order",
+        "sale_order_id",
+        string="Purchase Orders",
+        help="Purchase orders created from this sale order",
+    )
+    purchase_count = fields.Integer(
+        string="Purchase Orders",
+        compute="_compute_purchase_count",
+        help="Number of purchase orders created from this sale order",
+    )
+
+    @api.depends("purchase_order_ids")
+    def _compute_purchase_count(self):
+        """
+        Compute the number of RFQs created from this sale order
+        """
+        for order in self:
+            order.purchase_count = len(order.purchase_order_ids)
+
+    def action_view_purchase_orders(self):
+        """
+        View the purchase orders created from this sale order
+        """
+        self.ensure_one()
+        action = self.env.ref("purchase.purchase_rfq").read()[0]
+        action["domain"] = [("id", "in", self.purchase_order_ids.ids)]
+        return action
+
+    def action_create_purchase_orders(self):
+        """
+        Create purchase orders from this sale order
+        """
+        self.ensure_one()
+        self._create_purchase_orders()
+        if self.purchase_count:
+            return self.action_view_purchase_orders()
+        raise UserError(_("No purchase orders created"))
+
+    def _get_capable_vendors(self, lead_time=0):
+        """
+        Returns a recordset of res.partner records that match the criteria
+
+        param lead_time: Integer of days to match
+
+        return: Recordset of res.partner
+        """
+        self.ensure_one()
+        vendors_dict = self._get_vendor_dict(lead_time)
+        return vendors_dict.keys()
+
+    def _create_purchase_orders(self):
+        """
+        Create purchase orders from this sale order
+        """
+        self.ensure_one()
+        vendors_dict = self._get_vendor_dict()
+        for vendor, value in vendors_dict.items():
+            vals_purchase = {
```

Comment:
In order to make this method easier to override later, I suggest you to move values preparation into a separate method
```python
purchase = self.env["purchase.order"].create(
    self._prepare_purchase_order_vals(vendor, lines)
)
```